data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-api/src/main/java/org/eclipse/edc/protocol/dsp/transferprocess/api/controller/DspTransferProcessApiController.java

```diff
@@ -176,17 +222,32 @@ public void transferProcessCompletion(@PathParam("id") String id, JsonObject jso
      * @param id         the ID of the process
      * @param jsonObject the {@link TransferTerminationMessage} in JSON-LD expanded form
      * @param token      the authorization header
+     * @return empty response or error.
      */
     @POST
     @Path("{id}" + TRANSFER_TERMINATION)
-    public void transferProcessTermination(@PathParam("id") String id, JsonObject jsonObject, @HeaderParam(AUTHORIZATION) String token) {
-        handleMessage(MessageSpec.Builder.newInstance(TransferTerminationMessage.class)
-                .processId(id)
-                .expectedMessageType(DSPACE_TRANSFER_TERMINATION_TYPE)
-                .message(jsonObject)
-                .token(token)
-                .serviceCall(protocolService::notifyTerminated)
-                .build());
+    public Response transferProcessTermination(@PathParam("id") String id, JsonObject jsonObject, @HeaderParam(AUTHORIZATION) String token) {
+        try {
+            handleMessage(MessageSpec.Builder.newInstance(TransferTerminationMessage.class)
+                    .processId(id)
+                    .expectedMessageType(DSPACE_TRANSFER_TERMINATION_TYPE)
+                    .message(jsonObject)
+                    .token(token)
+                    .serviceCall(protocolService::notifyTerminated)
+                    .build());
+        } catch (AuthenticationFailedException exception) {
+            monitor.debug(String.format(UNAUTHORIZED + "Requested process id %s: %s", id, exception.getMessages()));
+            return errorResponse(Optional.of(id), Response.Status.UNAUTHORIZED, UNAUTHORIZED);
+        } catch (InvalidRequestException exception) {
+            monitor.debug(String.format(BAD_REQUEST + "Requested process id %s: %s", id, exception.getMessages()));
+            return errorResponse(Optional.of(id), Response.Status.BAD_REQUEST, BAD_REQUEST);
+        } catch (Exception exception) {
+            var errorCode = UUID.randomUUID();
```

Comment:
that seems spurious: we should not catch just _any_ exception. As discussed, only errors on the egress path (e.g. during transformation, compaction,...) should actually get return as an "explicit" 50x. Unexpected exceptions should just get bubbled up the stacktrace and be logged.